Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm

```diff
@@ -3415,48 +3415,55 @@ HTTPServer server([&] (Connection connection) {
 onload = async () => {
     try {
         // Create cache1
-        cache1 = await self.caches.open("test1");
-        await cache1.put(new Request('test'), new Response('my response'));
+        cache1 = await self.caches.open("cache1");
+        await cache1.put(new Request('my request'), new Response('my response'));
         webkit.messageHandlers.sw.postMessage("PASS");
     } catch(e) {
-        webkit.messageHandlers.sw.postMessage("fill failed with " + e);
+        webkit.messageHandlers.sw.postMessage("put failed with " + e);
     }
 }
 
-function check()
-{
-    // Create several caches, one of which should have the same ObjectIdentifier value as cache1.
-    Promise.all([
-        self.caches.open("test2"),
-        self.caches.open("test3"),
-        self.caches.open("test4"),
-        self.caches.open("test5"),
-        self.caches.open("test6"),
-        self.caches.open("test7"),
-        self.caches.open("test8"),
-        self.caches.open("test9"),
-        self.caches.open("test10")
-    ]).then(() => {
-        cache1.match('test').then(() => {
-            webkit.messageHandlers.sw.postMessage("cache1 did match unexpectedly");
+var nextCacheNumber = 2;
+function match()
+{
+    return new Promise((resolve, reject) => {
+        self.caches.open("cache" + nextCacheNumber++).then(() => {
+            cache1.match('my request').then(() => {
```

Comment:
`match` should not be in the loop based on what the comment below said.
It should probably be something like:
```
// Open
const maxAttempt = 10;
let count  = 0;
let newCache = null;
while (count < maxAttempt) {
    try {
       newCache = await self.caches.open(cacheName);
    } catch(e) {
       ++count;
    }
}

if (!newCache) {
    sendFailureMessage();
    return;
}

// Match
cache1.match()...
```

Reply:
Good suggestion--the code does look nicer that way. I've updated and uploaded it again.

But why do we want an attempt limit? This would make the test non-deterministic, which is exactly the problem we're trying to solve. Isn't it better to keep looping until the open call succeeds? That way, the test won't be flaky.

Reply:
Based on what you showed me, I think the limit is not what makes the test non-deterministic, but original test does not expect all `open()` calls to succeed? 
We usually have a max attempt in test because we don't want test to keep trying until timeout. First, normally it does not need 30s to prove something is wrong. Second it would be hard to debug timeout (we won't know if it's because some promise does not resolve or because of the loop). Third timeout would slow down the whole test run.
If you have concern, we could start with some big number like 50.

Reply:
It is true that the original test does not expect all open() calls to succeed. But it's possible that all of the 10 calls fail since there's no guarantee on when the web process is notified--so the test would still be non-deterministic. (These are the two issues we're trying to fix). 

So I thought this loop would be good since it runs exactly the number of open() calls to make the test succeed and locally it has been running quickly (<1 second).

But I did put the limit in since you mentioned that we don't want the test to keep trying until timeout in case something is wrong. And I added a comment so that should the test show up as flaky again, we know to increase the number of attempts :)